plugin/src/main/java/org/opennms/velocloud/rest/v1/VelocloudRestServiceV1Impl.java

```diff
@@ -52,4 +60,81 @@ public List<EnterpriseDTO> getCustomersForMspPartner(final String alias) throws
                      .map(customer -> Mapper.ENTERPRISE_INSTANCE.sourceToTarget(customer))
                      .collect(Collectors.toList());
     }
+
+    @Override
+    public List<ConnectionListElementDTO> getConnectionList() {
+        return this.connectionManager.getAliases().stream()
+                .map(alias -> {
+                    var connection = connectionManager.getConnection(alias).orElseThrow();
+                    final var connectionDTO = new ConnectionListElementDTO();
+                    connectionDTO.setAlias(alias);
+                    connectionDTO.setOrchestratorUrl(connection.getOrchestratorUrl());
+                    connectionDTO.setApiKey("*******");
+                    return connectionDTO;
+                })
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public Response addConnection(String alias, boolean dryRun, ConnectionDTO connectionDTO) throws VelocloudApiException {
+        if (this.connectionManager.getAliases().contains(alias)) {
+            return Response.status(Response.Status.BAD_REQUEST).entity("Connection already exists").build();
+        }
+        try {
+            clientManager.validate(VelocloudApiClientCredentials.builder()
+                    .withOrchestratorUrl(connectionDTO.getOrchestratorUrl())
+                    .withApiKey(connectionDTO.getApiKey())
+                    .build());
+        }
+        catch (VelocloudApiException e) {
+            return Response.status(Response.Status.BAD_REQUEST).entity("Invalid credentials").build();
```

Comment:
Not sure but looks to me that if the connection fails because the other side is unreachable this exeptions won't mention that.